UM/Operations/OperationStack.py

```diff
@@ -16,6 +17,17 @@ def __init__(self):
         self._current_index = -1 #Index of the most recently executed operation.
         self._lock = threading.Lock() #Lock to make sure only one thread can modify the operation stack at a time.
 
+        self._merge_operations = False  #Only merge operations when we're in the same operation.
+        self._controller = UM.Application.getInstance().getController()  # Circular dependency blah
+        self._controller.toolOperationStarted.connect(self._onToolOperationStarted)
+        self._controller.toolOperationStopped.connect(self._onToolOperationStopped)
+
+    def _onToolOperationStarted(self, tool):
+        self._merge_operations = False
+
+    def _onToolOperationStopped(self, tool):
+        self._merge_operations = False
```

Comment:
The first one should be True, I guess.
Personally I would name the variable ```self._operation_active``` or something similar, so when you use it in line 116 you know _why_ the operation is being merged.